src/public-lib/lib/persistantApiClientCache.ts

```diff
@@ -0,0 +1,52 @@
+type NameMethodPair = {
+    name: string,
+    method: any,
+}
+
+function createCacheCheckingVersionOfGet(pair: NameMethodPair, api: any): NameMethodPair {
+    const method = function(parameters: any) {
```

Comment:
the interceptor is going to need to return a promise whether the data exists or not, since that's the original method's response type.    And it will need to intercept the response of the original method in order to put the response in cache.